apps/aecore/src/aec_chain.erl

```diff
@@ -593,3 +594,58 @@ get_key_header_by_height(Height) when is_integer(Height), Height >= 0 ->
         error -> {error, chain_too_short};
         {ok, Hash} -> get_header(Hash)
     end.
+
+%%%===================================================================
+%%% Statistics
+%%%===================================================================
+
+-spec sum_tokens_at_height(aec_blocks:height()) ->
+                                  {error, 'chain_too_short'}
+                                | {ok, #{ 'accounts'         => non_neg_integer()
+                                        , 'contracts'        => non_neg_integer()
+                                        , 'contract_oracles' => non_neg_integer()
+                                        , 'locked'           => non_neg_integer()
+                                        , 'oracles'          => non_neg_integer()
+                                        , 'oracle_queries'   => non_neg_integer()
+                                        , 'pending_rewards'  => non_neg_integer()
+                                        , 'total'            => non_neg_integer()
+                                        }}.
+
+
+sum_tokens_at_height(Height) ->
+    %% Wrap in transaction for speed.
+    %% TODO: This could be done dirty
+    aec_db:ensure_transaction(fun() -> int_sum_tokens_at_height(Height) end).
+
+int_sum_tokens_at_height(Height) ->
+    case aec_chain_state:get_key_block_hash_at_height(Height) of
+        error -> {error, chain_too_short};
+        {ok, Hash} ->
+            {ok, Trees} = get_block_state(Hash),
+            Sum = aec_trees:sum_total_coin(Trees),
+            Sum1 = Sum#{pending_rewards => sum_pending_rewards(Height, Hash)},
+            Total = maps:fold(fun(_, X, Acc) -> Acc + X end, 0, Sum1),
+            {ok, Sum1#{ total => Total }}
+    end.
+
+sum_pending_rewards(0,_Hash) ->
+    0;
+sum_pending_rewards(Height, Hash) when Height > 0 ->
+    %% Rewards for a generation is given for the closing of the generation,
+    %% so we need stop at the closing key block (hence the +1).
+    StopHeight = max(Height - aec_governance:beneficiary_reward_delay() + 1, 1),
+    sum_pending_rewards(Hash, StopHeight, 0).
+
+sum_pending_rewards(Hash, StopHeight, Acc) ->
+    {ok, Header}  = get_header(Hash),
+    Height        = aec_headers:height(Header),
+    {value, Fees} = aec_db:find_block_fees(Hash),
+    Coinbase      = aec_coinbase:coinbase_at_height(Height),
+    Acc1 = Acc + Fees + Coinbase,
```

Comment:
This seems not to cater for fraud already-reported though pending for delay. Could be considered as optional enhancement. (I need to review this logic.)

Reply:
Agreed, but since this should be a pretty rare case, I don' think we should spend effort in this.

Reply:
In that case it would be good to mention that assumption in a comment.

Reply:
The pending rewards is always pending until the beneficiary reward delay has passed. Then the decision is made whether to pay out the reward, or to lock the reward and pay a smaller reward to the reporter of the fraud. Given this, the semantics of the summation made here is intact.

It would be, as Luca says, an improvement to specify this as a separate field, and there are other things that would be enhancing the information as well (e.g., differing between fees and coinbase, differing between deposits and available funds in contract). I just wanted to bite off at some level.

The fraud reporting is there to deter fraud, and we are mostly not expecting this to happen. That is why I don't want to spend more CPU cycles to find this information in this code.

Reply:
Thanks for the explanation. I agree with your reasoning.